### PR TITLE
Refactored search to handle custom field weights

### LINKED ISSUE #76: Query Interpretation: Custom Search Ranking
We'd like for frontend to be able to pass its own search ranking weightings and options such that backend can take in these rankings arbitrarily and sort based on these criteria.

## Context from Evan's Design Docs
There are two ways to interpret a given query. 

- Typing in query and hitting go
- Typing in query and clicking on a suggestion

These cases will be handled the same way, the only difference is that the suggestions have an attribute to it (name, point of contact, or tag) and by using the suggestion it will automatically bring that search case to the top. 

**Attributes:**

- Business Name
- Point of contact
- Tag

**Example:**

I search for *"Angela"* and click on the suggestion *"Angela King"* with the point of contact attribute. 

So when this happens, we return the resource card with the point of contact *"Angela King"* with the rest of the search results below. So the resulting rank of the search algorithm would have the weighted result with the point of contact 

1. Weighted results w/ point of contact weighted at 1.0 and everything else at 0
2. Normal weighted results

Note that point of contact is on the top of the search ranking now. Only the order changes. The internal rankings for these search categories stay the same. 

## Rankings Overview

So we know that there are different categories with internal rankings and parameters. All their weighted scores will add up to 1.0 . These are:

1. Name of business - **0.3**
2. Point of contact - **0.35**
3. Tags - **0.15**
4. Matches in Description - **0.1**
5. Matches in Notes - **0.1**

These weights will all have an internal score out of 1.0 also, so that when you multiply the scores by their weight, you can get a max score of 1.0.

### PR DESCRIPTION
**Pull request by alicesf2**
## Status:

:construction: In development

## Description
Added an optionalParams query param to handle custom rankings
Fixes #76

### CODE REVIEW

File: backend/routes/api/resources.js [hunk 1]
```diff
@@ -49,6 +51,9 @@ router.get(
       });
     }
     if (keyword) {
+      if (optionalParams) {
+        options.keys = optionalParams;
```
Comment (alan-fang) [inline backend/routes/api/resources.js]: add a comment detailing what this does:
`// something about weighted rankings`

File: backend/routes/api/resources.js [hunk 2]
```diff
@@ -28,11 +29,12 @@ router.get(
...
       lat: Joi.number(),
       long: Joi.number(),
-      keyword: Joi.string()
+      keyword: Joi.string(),
+      optionalParams: Joi.array()
     }
   }),
   errorWrap(async (req, res) => {
-    const { radius, lat, long, keyword } = req.query;
+    const { radius, lat, long, keyword, optionalParams } = req.query;
```
Comment (alan-fang) [inline backend/routes/api/resources.js]: maybe call it something more descriptive?

File: backend/routes/api/resources.js [hunk 3]
```diff
@@ -28,11 +29,12 @@ router.get(
       radius: Joi.number(),
       lat: Joi.number(),
       long: Joi.number(),
-      keyword: Joi.string()
+      keyword: Joi.string(),
+      optionalParams: Joi.array()
```
Comment (alan-fang) [inline backend/routes/api/resources.js]: this should also be `customWeights` right

Comment (alicesf2) [reply backend/routes/api/resources.js]: oop u right

**Pull request merged**